app/src/main/java/org/schabi/newpipe/settings/notifications/NotificationsConfigAdapter.kt

```diff
@@ -0,0 +1,111 @@
+import androidx.vectordrawable.graphics.drawable.Animatable2Compat
+import androidx.vectordrawable.graphics.drawable.AnimatedVectorDrawableCompat
+import org.schabi.newpipe.R
+import org.schabi.newpipe.database.subscription.NotificationMode
+import org.schabi.newpipe.database.subscription.SubscriptionEntity
+import org.schabi.newpipe.settings.notifications.NotificationsConfigAdapter.SubscriptionHolder
+import org.schabi.newpipe.util.ThemeHelper
+import java.util.*
+
+internal class NotificationsConfigAdapter(
+        private val listener: ModeToggleListener
+) : RecyclerView.Adapter<SubscriptionHolder>() {
+
+    private val dataSet = ArrayList<SubscriptionEntity>()
+
+    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): SubscriptionHolder {
+        val view = LayoutInflater.from(viewGroup.context)
+                .inflate(R.layout.item_notification_config, viewGroup, false)
+        return SubscriptionHolder(view, listener)
+    }
+
+    override fun onBindViewHolder(subscriptionHolder: SubscriptionHolder, i: Int) {
+        subscriptionHolder.bind(dataSet[i])
+    }
+
+    fun getItem(position: Int) = dataSet[position]
+
+    override fun getItemCount() = dataSet.size
+
+    fun update(newData: List<SubscriptionEntity>?) {
+        dataSet.clear()
+        dataSet.addAll(newData!!)
+        notifyDataSetChanged()
+    }
+
+    internal class SubscriptionHolder(
+            itemView: View,
+            private val listener: ModeToggleListener
+    ) : RecyclerView.ViewHolder(itemView), View.OnClickListener, ModeToggleListener {
+
+        private val titleTextView: TextView = itemView.findViewById(R.id.text_title)
+        private val stateImageView: ImageView = itemView.findViewById(R.id.icon_state)
+
+        init {
+            stateImageView.setOnClickListener(this)
+        }
+
+        fun bind(data: SubscriptionEntity) {
+            titleTextView.text = data.name
+            stateImageView.setImageDrawable(getIcon(data.notificationMode))
+        }
+
+        override fun onClick(v: View) {
+            val icon = stateImageView.drawable as Animatable
+            if (!icon.isRunning) {
+                icon.start()
+            }
+        }
+
+        override fun onModeToggle(position: Int, mode: Int) {
+            listener.onModeToggle(adapterPosition, mode)
+            stateImageView.setImageDrawable(getIcon(mode))
+        }
+
+        private fun getIcon(@NotificationMode mode: Int): Drawable? {
+            val icon = AnimatedVectorDrawableCompat.create(
+                    itemView.context,
+                    if (mode == NotificationMode.DISABLED) {
+                        R.drawable.av_notification_off
+                    } else {
+                        R.drawable.av_notification_on
+                    }
+            ) ?: return null
+            icon.setTint(ThemeHelper.resolveColorFromAttr(
+                    itemView.context, android.R.attr.textColorPrimary
+            ))
```

Comment:
You should use an `attr` icon here, so that it can be styled according to themes automatically. Also, why is the name of those icons not `ic_notification_on_white_24dp`? So, in the end you should have 4 icons (`ic_notification_on_white_24dp`, `ic_notification_on_black_24dp`, `ic_notification_off_white_24dp`, `ic_notification_off_black_24dp`)

Reply:
What reasons to use different icons if we can use one icon with tint?